api/src/main/java/org/pmiops/workbench/db/model/DbInstitution.java

```diff
@@ -78,21 +94,69 @@ public void setOrganizationTypeOtherText(String organizationTypeOtherText) {
     this.organizationTypeOtherText = organizationTypeOtherText;
   }
 
+  /** Builder method to help streamline the building of a DbInstitution. */
+  public DbInstitution organizationTypeOtherText(String organizationTypeOtherText) {
+    this.organizationTypeOtherText = organizationTypeOtherText;
+    return this;
+  }
+
+  @OneToMany(mappedBy = "institution", cascade = CascadeType.ALL, fetch = FetchType.EAGER)
   public Set<DbInstitutionEmailDomain> getEmailDomains() {
     return emailDomains;
   }
 
-  public void setEmailDomains(Iterable<DbInstitutionEmailDomain> emailDomains) {
-    this.emailDomains =
-        StreamSupport.stream(emailDomains.spliterator(), false).collect(Collectors.toSet());
+  /**
+   * Hibernate doesn't like it when you reassign collections. Instead, modify in-place. First, call
+   * retainAll() to subset DB rows to those we wish to keep: the intersection of old and new. Then,
+   * call addAll() to add the diff(new - old) rows.
+   *
+   * <p>https://stackoverflow.com/questions/5587482/hibernate-a-collection-with-cascade-all-delete-orphan-was-no-longer-referenc
+   *
+   * @param emailDomains the new set of domains for this Institution
+   */
+  public void setEmailDomains(final Set<DbInstitutionEmailDomain> emailDomains) {
+    final Set<DbInstitutionEmailDomain> attachedDomains =
+        emailDomains.stream()
+            .map(address -> address.setInstitution(this))
+            .collect(Collectors.toSet());
+    // modifies this set so that its value is the intersection of the two sets
+    this.emailDomains.retainAll(attachedDomains);
+    this.emailDomains.addAll(Sets.difference(attachedDomains, this.emailDomains));
   }
 
+  /** Builder method to help streamline the building of a DbInstitution. */
+  public DbInstitution emailDomains(Set<DbInstitutionEmailDomain> emailDomains) {
+    setEmailDomains(emailDomains);
+    return this;
+  }
+
+  @OneToMany(mappedBy = "institution", cascade = CascadeType.ALL, fetch = FetchType.EAGER)
   public Set<DbInstitutionEmailAddress> getEmailAddresses() {
     return emailAddresses;
   }
 
-  public void setEmailAddresses(Iterable<DbInstitutionEmailAddress> emailAddresses) {
-    this.emailAddresses =
-        StreamSupport.stream(emailAddresses.spliterator(), false).collect(Collectors.toSet());
+  /**
+   * Hibernate doesn't like it when you reassign collections. Instead, modify in-place. First, call
```

Comment:
Can you explain at a higher level what this function is trying to accomplish? That is, why are we doing multiple set intersections? And why do we need a set difference if emailAddresses is a set to begin with? I thought adding an existing item was a no-op.

Reply:
Added a comment.  Effectively it is doing a simple set assignment.

For the last step, we could add the difference or the whole set again.  Should be identical.